vertx/src/main/java/com/microsoft/gctoolkit/vertx/jvm/AbstractJavaVirtualMachine.java

```diff
@@ -32,95 +29,112 @@
     }
 
     @Override
     public boolean isSerial() {
-        return jvmConfigurationFromParser.getDiary().isSerialFull();
+        return diary.isSerialFull();
     }
 
     @Override
     public boolean isCMS() {
-        return jvmConfigurationFromParser.getDiary().isCMS();
+        return diary.isCMS();
     }
 
     @Override
     public String getCommandLine() {
-        return jvmConfigurationFromParser.getCommandLine();
+        return ""; //todo: extract from diary... jvmConfigurationFromParser.getCommandLine();
     }
 
+    /**
+     * of the first event is significantly away from zero in relation to the time intervals between the
+     * of the next N events, where N maybe 1.
+     *
+     * try to estimate the time at which the JVM started. For log fragments, this will be the time
+     * of the first event in the log. Otherwise it will be 0.000 seconds.
+     * @return DateTimeStamp
+     */
+    @Override
+    public DateTimeStamp getEstimatedJVMStartTime() {
+        DateTimeStamp startTime = getTimeOfFirstEvent();
+        // Initial entries in GC log happen within seconds. Lets allow for 60 before considering the log
+        // to be a fragment.
+        if (startTime.getTimeStamp() < LOG_FRAGMENT_THRESHOLD) {
+            return startTime.minus(startTime.getTimeStamp());
+        } else {
+            return startTime;
+        }
+    }
+
+    /**
+     * todo: fix this to be a globally available value. Also, the JVM start time is not zero if the time
+     * of the first event is significantly away from zero in relation to the time intervals between the
+     * of the next N events, where N maybe 1.
+     *
+     * try to estimate the time at which the JVM started. For log fragments, this will be the time
+     * of the first event in the log. Otherwise it will be 0.000 seconds.
+     * @return DateTimeStamp
+     */
     @Override
     public DateTimeStamp getTimeOfFirstEvent() {
-        return jvmConfigurationFromParser.getTimeOfFirstEvent();
+        return diary.getTimeOfFirstEvent();
     }
 
+    /**
+     * JVM termination time will be one of either, the time stamp in the termination event if present or, the
+     * time of the last event + that events duration.
+     * @return DateTimeStamp
+     */
     @Override
-    public DateTimeStamp getTimeOfLastEvent() {
-        if (getTimeOfFirstEvent().before(timeOfLastEvent))
-            return timeOfLastEvent;
-        else
-            return getTimeOfFirstEvent();
+    public DateTimeStamp getJVMTerminationTime() {
+        return timeOfLastEvent;
     }
 
     @Override
     public double getRuntimeDuration() {
-        boolean isLogFragment = getTimeOfFirstEvent().getTimeStamp() > LOG_FRAGMENT_THRESHOLD;
-        if (isLogFragment)
-            return getTimeOfLastEvent().minus(getTimeOfFirstEvent());
-        return getTimeOfLastEvent().getTimeStamp();        }
+        return getJVMTerminationTime().minus(getEstimatedJVMStartTime());
```

Comment:
IIRC, the comment in the interface says that runtime duration is not necessarily end time minus start time. Check the interface and see if that comment needs to be fixed.

Reply:
I believe the comment is in a correct state